### PR TITLE
add test case for Manager and WitnessProductBlockService

### PR DESCRIPTION
**Pull request by spidemen**
**What does this PR do?**
1. add test  cases for Manager.java
2. create test  cases for WitnessProductBlockService.java

**Why are these changes required?**
    increase code coverage 
**This PR has been tested by:**
- Unit Tests
- Manual Testing

**Follow up**

**Extra details**

### CODE REVIEW

File: framework/src/test/java/org/tron/core/db/ManagerTest.java [hunk 1]
```diff
@@ -47,25 +58,37 @@
...
   private static BlockCapsule blockCapsule2;
   private static String dbPath = "output_manager_test";
   private static AtomicInteger port = new AtomicInteger(0);
+  private static String accountAddress =
+      Wallet.getAddressPreFixString() + "548794500882809695a8a687866e76d4271a1abc";
 
   @Before
   public void init() {
-    Args.setParam(new String[]{"-d", dbPath, "-w"}, Constant.TEST_CONF);
+    Args.setParam(new String[] {"-d", dbPath, "-w"}, Constant.TEST_CONF);
```
Comment (shydesky) [inline framework/src/test/java/org/tron/core/db/ManagerTest.java]: remove blank

File: framework/src/test/java/org/tron/core/db/ManagerTest.java [hunk 2]
```diff
@@ -177,9 +201,301 @@ public void pushBlock() {
...
+    dbManager.getDynamicPropertiesStore().saveAllowSameTokenName(1);
+    Assert.assertTrue(dbManager.getExchangeStoreFinal() instanceof ExchangeV2Store);
+
+  }
+
+
+  @Test
+  public void pushBlockInvalidSignature() {
+    // invalid witness address cause invalid signature
+    String invalideWitness = "bcab94c3e0c9214fb4ac7ff9d7d5a937d1f40031f";
```
Comment (shydesky) [inline framework/src/test/java/org/tron/core/db/ManagerTest.java]: invalidWitness

File: framework/src/test/java/org/tron/core/db/ManagerTest.java [hunk 3]
```diff
@@ -177,9 +201,301 @@ public void pushBlock() {
...
+    blockCapsule2.setWitness(invalideWitness);
+    try {
+      dbManager.pushBlock(blockCapsule2);
+      Assert.assertTrue(false);
+    } catch (BadBlockException e) {
+      Assert.assertTrue(e instanceof BadBlockException);
```
Comment (shydesky) [inline framework/src/test/java/org/tron/core/db/ManagerTest.java]: unused, always true

File: framework/src/test/java/org/tron/core/db/ManagerTest.java [hunk 4]
```diff
@@ -177,9 +201,301 @@ public void pushBlock() {
...
+    }
+
+
+    account.setBalance(30);
+    chainManager.getAccountStore().put(account.createDbKey(), account); // update balance
+    try {
+      dbManager.adjustBalance(accountAddress.getBytes(), -40);
+      Assert.assertTrue(false);
+    } catch (BalanceInsufficientException e) {
+      Assert.assertTrue(e instanceof BalanceInsufficientException);
```
Comment (shydesky) [inline framework/src/test/java/org/tron/core/db/ManagerTest.java]: remove this unused judgment code

File: framework/src/test/java/org/tron/core/db/ManagerTest.java [hunk 5]
```diff
@@ -313,8 +629,8 @@ public void doNotSwitch()
       TaposException, BadNumberBlockException, NonCommonBlockException,
       ReceiptCheckErrException, VMIllegalException, TooBigTransactionResultException,
       ZksnarkException {
-    Args.setParam(new String[]{"--witness"}, Constant.TEST_CONF);
-    long size = dbManager.getBlockStore().size();
+    Args.setParam(new String[] {"--witness"}, Constant.TEST_CONF);
```
Comment (shydesky) [inline framework/src/test/java/org/tron/core/db/ManagerTest.java]: remove blank

File: framework/src/test/java/org/tron/core/db/ManagerTest.java (hunk 2 above)
Comment (spidemen) [reply framework/src/test/java/org/tron/core/db/ManagerTest.java]: ok

File: framework/src/test/java/org/tron/core/db/ManagerTest.java (hunk 3 above)
Comment (spidemen) [reply framework/src/test/java/org/tron/core/db/ManagerTest.java]: will remove it

File: framework/src/test/java/org/tron/core/db/ManagerTest.java (hunk 5 above)
Comment (spidemen) [reply framework/src/test/java/org/tron/core/db/ManagerTest.java]: ok

**Pull request merged**